File: src/main/java/com/oracle/ocmjea/gof/behavioral/visitor/Computer.java
package com.oracle.ocmjea.gof.behavioral.visitor;

public class Computer implements ComputerPart {

	ComputerPart[] parts;

	public Computer() {
		parts = new ComputerPart[] { new Mouse(), new Monitor(), new Teclado() };
	}

	@Override
	public void accept(ComputerPartVisitor computerPartVisitor) {
		for (int i = 0; i < parts.length; i++) {
			parts[i].accept(computerPartVisitor);
		}
		computerPartVisitor.visit(this);
	}
}

File: design-patterns/src/main/kotlin/designpatterns/behavioral/visitor/Computer.kt
package designpatterns.behavioral.visitor

class Computer : ComputerPart {
    private val parts: Array<ComputerPart> = arrayOf(Mouse(), Monitor(), Keyboard())

    override fun accept(computerPartVisitor: ComputerPartVisitor) {
        parts.forEach { it.accept(computerPartVisitor) }
        computerPartVisitor.visit(this)
    }
}
